ttnn/cpp/ttnn/operations/moreh/moreh_matmul/device/moreh_matmul_program_factory.cpp

```diff
@@ -39,7 +39,7 @@ std::vector<int64_t> find_reduce_dim(const tt::tt_metal::LegacyShape &a_shape, c
     // batch dims
     for (int i = 0; i < rank - 2; ++i) {
         int idx = rank - 1 - i;
-        TT_ASSERT(idx >= 0);
+        TT_ASSERT(idx >= 0, "idx < 0");
```

Comment:
```suggestion
        TT_FATAL(idx >= 0, "idx < 0");
```

Reply:
I'd argue that this is out-of-scope for this PR, so I've put it in a separate commit